### PR TITLE
Prevent Division of zero issues and fix restyler

### PR DESCRIPTION
**Pull request by Elias2660**
In the edge case, when the model collapses. 

Screenshot 2025-03-20 at 08 32 00

Also, I think we should shift down the learning rate. The model collapsed on the first epoch in a relatively small training set.

### CODE REVIEW

File: utility/eval_utility.py [hunk 1]
```diff
@@ -216,7 +216,7 @@ def accuracy(self):
         In a multi-label system a prediction is only counted as accurate if *all* labels match, so
         accuracy in multi-label systems should be expected to be lower.
         """
-        return self.correct_count / self.prediction_count
+        return self.correct_count / (self.prediction_count + 1e-20)
```
Comment (bfirner) [inline utility/eval_utility.py:236]: Could you make the `epsilon` variable an argument with a default value, as is done in the `normalizeImages` function in `utility/train_utility.py`? I think that exposes enough of the internals that we can adjust it for different applications if necessary.

Comment (Elias2660) [reply utility/eval_utility.py]: done

File: utility/eval_utility.py [hunk 2]
```diff
@@ -86,31 +98,38 @@ def __init__(self, size, units = None, names = None):
...
 
     def labelMeans(self):
+        """ """
         return [stats.mean() for stats in self.prediction_statistics]
 
     def update(self, predictions, labels):
-        """ Update the statistics matrix with a new set of predictions and labels.
+        """Update the statistics matrix with a new set of predictions and labels.
+
+        :param predictions: torch
```
Comment (bfirner) [inline utility/eval_utility.py:130]: Which formatter is reformatting all of the function comments? Is this matching some doc builder? It's doing a terrible job, turning the `torch.tensor` type into the meaningless `torch` and removing the rest of the docstring. I don't like it.

Comment (bfirner) [reply utility/eval_utility.py:130]: Looks like it's Pyment. It sucks. It makes the documentation less human readable so that it's better for Sphinx or some other autodoc generator, but that's backwards -- the comments should be human readable and the autodoc generators should work around that. Unless you have some attachment to it, I vote we don't use it and remove its commit from this PR.

Comment (Elias2660) [reply utility/eval_utility.py:130]: ok, that's good with me

Comment (Elias2660) [reply utility/eval_utility.py:130]: I'll look for a way to remove it from the workflow

**Comment by Elias2660** [comment]
@bfirner  fixed the restyler and reverted the commits

**Comment by Elias2660** [comment]
going to merge by the end of the day if there are no other issues

**Comment by bfirner** [comment]
Looks good to me. I'll trigger the merge.
